File: src/python/data_models/conversation_data.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

class ConversationTurn:
    def __init__(
        self,
        role: str = None,
        content: str = None
    ):
        self.role = role
        self.content = content

    def toJSON(self):
        return {"role": self.role, "content": self.content}

class ConversationData:
    def __init__(
        self,
        history: list[ConversationTurn],
        max_turns: int = 10,
        thread_id: str = None,
    ):
        self.thread_id = thread_id
        self.history = history
        self.max_turns = max_turns

    def add_turn(self, role: str, content: str):
        self.history.append(ConversationTurn(role, content))
        if len(self.history) >= self.max_turns:
            self.history.pop(0)

    def toMessages(self):
        return [turn.toJSON() for turn in self.history]

File: src/node/data_models/conversation_data.js
class ConversationTurn {
    constructor(role, message, imageType, imageData) {
        this.role = role;
        this.message = message;
        this.imageType = imageType;
        this.imageData = imageData;
    }
}

class ConversationData {
    constructor(history, max_turns = 10, thread_id = null) {
        this.thread_id = thread_id;
        this.history = history;
        this.max_turns = max_turns;
        this.attachments = [];
    }

    static addTurn(conversationData, role, message, imageType, imageData) {
        conversationData.history.push(new ConversationTurn(role, message, imageType, imageData));
        if (conversationData.history.length >= conversationData.max_turns) {
            conversationData.history.splice(1, 1);
        }
    }

    static toMessages(conversationHistory) {
        let messages = conversationHistory.map((turn, index) => ({
            role: turn.role,
            tool_calls: turn.tool_calls,
            content: turn.message
        }))
        return messages

    }
    
}

module.exports = { ConversationTurn, ConversationData };